machine/os/ubuntu.go

```diff
@@ -0,0 +1,37 @@
+package os
+
+import (
+	"strings"
+
+	"github.com/docker/machine/drivers"
+)
+
+func init() {
+	RegisterRuntime("ubuntu", &RegisteredRuntime{
+		Detect: UbuntuDetection,
+	})
+}
+
+func UbuntuDetection(d drivers.Driver) (*Runtime, error) {
+	out, err := sshCommand(d, "lsb_release -i | cut -f2 -d:")
```

Comment:
@sthulb  The success or failure of running commands like `'cat /etc/os-release | egrep ^ID=boot2docker'` is probably what we're after, then.  I have to admit it's pretty tempting to just run `cat /etc/os-release | egrep ^ID` once and then do a pattern match on the output once instead of running a command over SSH for each possible driver though :)